app/controllers/application_controller.rb

```diff
@@ -92,5 +92,11 @@ def rate_limit!(action)
     rate_limiter.check_limit!(action)
   end
 
-  delegate :rate_limiter, to: :current_user
+  def rate_limiter
+    (current_user || anonymous_user).rate_limiter
+  end
+
+  def anonymous_user
```

Comment:
:fire: Finally, a proper null object! 😍 I'm a big fan of this pattern and would like if we used this more widely instead of `nil` checks on `current_user` as it has the potential of removing many conditionals and making the code more "tell, don't ask".

Reply:
Also: anonymous users are users :-)